File: src/main/java/api/ApiAccess.java
package api;

import lombok.AllArgsConstructor;
import lombok.Getter;

import javax.servlet.http.HttpServletRequest;
import java.util.Map;

/**
 * 调用 Api 时的一些信息，可以选择使用
 */
@Getter
@AllArgsConstructor
public class ApiAccess {
    /**
     * Jetty 提供的请求信息
     */
    private final HttpServletRequest request;
    /**
     * 请求头
     */
    private final Map<String, String> headers;
    /**
     * 请求内容
     */
    private final String content;
}

File: src/main/kotlin/org/hydev/ApiAccess.kt
package org.hydev

import org.eclipse.jetty.server.Request
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * An ApiAccess instance contains the data that the client sent to the server.
 *
 * @author HyDEV Team (https://github.com/HyDevelop)
 * @author Hykilpikonna (https://github.com/hykilpikonna)
 * @author Vanilla (https://github.com/VergeDX)
 * @since 2020-07-03 12:37
 */
class ApiAccess(
    target: String,
    baseRequest: Request,
    request: HttpServletRequest,
    response: HttpServletResponse,
    val body: String
): HttpAccess(target, baseRequest, request, response)
{
    val headers = request.mapHeaders()
}
